src/server/scripts/BrokenIsles/MawOfSouls/boss_ymiron_the_fallen_king.cpp

```diff
@@ -0,0 +1,706 @@
+
+        return true;
+    }
+
+private:
+    Unit* _caster;
+    int8 _remainingVisualsToSpawn;
+};
+
+// 193460 - Bane
+class spell_ymiron_the_fallen_king_bane_periodic : public SpellScript
+{
+    void OnPrecast() override
+    {
+        GetCaster()->m_Events.AddEventAtOffset(new YmironBanePrecastVisual(GetCaster(), YmironBanePrecastVisual::BANE_VISUAL_NUM), 0s);
+    }
+
+    void Register() override { }
+};
+
+// 193460 - Bane
+class spell_ymiron_the_fallen_king_bane_periodic_AuraScript : public AuraScript
+{
+    static constexpr float BANE_MISSILE_DIST_OFFSET = 2.272f;
+    static constexpr float BANE_MISSILE_DIST_BASE = 10.0f;
+    static constexpr float BANE_MISSILE_ANGLE_OFFSET = 0.75f;
+
+    static constexpr int8 BANE_MAX_TOTAL_TICKS = 4 + 4 + 14;
+
+    void SpawnBaneOrb(int8 tickNumber)
+    {
+        float dist = _distances[tickNumber - 1];
+        float angle = _angles[tickNumber - 1];
+        Position dest = { GetTarget()->GetPositionX() + dist * std::cos(angle), GetTarget()->GetPositionY() + dist * std::sin(angle), GetTarget()->GetPositionZ(), angle};
+        GetTarget()->CastSpell(dest, SPELL_BANE_MISSILE, true);
+    }
+
+    void OnAfterApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        for (int8 i = 0; i < BANE_MAX_TOTAL_TICKS; i++)
+        {
+            float angle = Position::NormalizeOrientation(GetCaster()->GetOrientation() + (i * BANE_MISSILE_ANGLE_OFFSET));
+            _angles.push_back(angle);
+            _angles.push_back(angle);
+
+            float dist = BANE_MISSILE_DIST_BASE + (i * BANE_MISSILE_DIST_OFFSET);
+            _distances.push_back(dist);
+            _distances.push_back(dist);
+        }
+        Trinity::Containers::RandomShuffle(_angles);
+        Trinity::Containers::RandomShuffle(_distances);
+    }
+
+    void OnPeriodic(AuraEffect const* aurEff)
+    {
+        if (aurEff->GetTickNumber() <= 2)
+        {
+            for (int8 i = 0; i < 4; i++)
+                SpawnBaneOrb(aurEff->GetTickNumber() + i);
+        }
+        else
+            SpawnBaneOrb(aurEff->GetTickNumber() + 8);
+    }
+
+    void OnAfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        GetTarget()->CastSpell(GetTarget(), SPELL_ARISE_FALLEN_ENABLER, true);
+    }
+
+    void Register() override
+    {
+        AfterEffectApply += AuraEffectApplyFn(spell_ymiron_the_fallen_king_bane_periodic_AuraScript::OnAfterApply, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY, AURA_EFFECT_HANDLE_REAL);
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_ymiron_the_fallen_king_bane_periodic_AuraScript::OnPeriodic, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+        AfterEffectRemove += AuraEffectRemoveFn(spell_ymiron_the_fallen_king_bane_periodic_AuraScript::OnAfterRemove, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY, AURA_EFFECT_HANDLE_REAL);
+    }
+private:
+    std::vector<float> _angles;
```

Comment:
These can be std::array<float, BANE_MAX_TOTAL_TICKS * 2>, RandomShuffle works on that